### PR TITLE
Split `FilterModal` components into separate files and folders

### PR DESCRIPTION
**Pull request by oisincoveney**
Splits the components within `FilterModal` into separate files and folders. I did this to help prepare the filter components for the SDK, where we will not want the modal but the content inside it. 

This is being done as the first step of separating the filter logic/content from the actual modal itself. There will be a follow-up PR that will use the content in an SDK-specific context.

Absolutely zero refactoring of the logic was done here. I only moved files around, added folders and `index.ts` files, and simplified imports. The tests should pass as normal.

### CODE REVIEW

File: frontend/src/metabase/querying/components/FilterModal/utils/filters.ts
Comment (oisincoveney) [inline frontend/src/metabase/querying/components/FilterModal/utils/filters.ts:1]: Moved to `TabPanelItem/segments.ts` and `TabPanelItem/columns.ts`

File: frontend/src/metabase/querying/components/FilterModal/FilterModal.styled.tsx
Comment (oisincoveney) [inline frontend/src/metabase/querying/components/FilterModal/FilterModal.styled.tsx:1]: *   `TabPanelRoot` → `TabPanel.styled.tsx`
*   `TabPanelItem` → `TabPanelItem.styled.tsx`
*   `TabsListSidebar` → `TabList.styled.tsx`

File: frontend/src/metabase/querying/components/FilterModal/FilterEmptyState.tsx
Comment (oisincoveney) [inline frontend/src/metabase/querying/components/FilterModal/FilterEmptyState.tsx:1]: This was called `SearchEmptyState` but _really_ should not have been called that, considering we have another component called that for our actual search.

File: frontend/src/metabase/querying/components/FilterContent/index.ts
Comment (oisincoveney) [inline frontend/src/metabase/querying/components/FilterContent/index.ts:1]: These are the only components used in `FilterModal` so didn't see a point in exporting everything

**Review by npretto — APPROVED** [review]

**Pull request merged**